src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -557,10 +562,38 @@ private void validateDeviceConfiguration(String thingName, String certificateFil
         }
     }
 
+    /**
+     * Validate the IoT credential and data endpoint with the provided AWS region. Currently it checks that the if the
+     * endpoints are provided, then the AWS region should be a part of the URL.
+     *
+     * @param awsRegion       the provided AWS region.
+     * @param iotCredEndpoint the provided IoT credentials endpoint
+     * @param iotDataEndpoint the providedIoT data endpoint
+     * @throws DeviceConfigurationException if the region is not valid or if the IoT endpoints do not have the AWS
+     *                                      region as a part of its URL.
+     */
+    public void validateEndpoints(String awsRegion, String iotCredEndpoint, String iotDataEndpoint)
+            throws DeviceConfigurationException {
+        // If the region value is empty/null, then try to get the region from the SDK lookup path
+        if (Utils.isNotEmpty(awsRegion) && !Region.regions().contains(Region.of(awsRegion))) {
+            logger.atWarn().log("Error looking up AWS region");
+            throw new DeviceConfigurationException(String.format("Error looking up AWS region %s", awsRegion));
+        }
+        if (Utils.isNotEmpty(iotCredEndpoint) && !iotCredEndpoint.contains(awsRegion)) {
+            throw new DeviceConfigurationException(String.format("Error setting up IoT credentials endpoint %s",
```

Comment:
this is far too generic. Tell the user what the problem is.

"IoT credential endpoint region does not match the AWS region of the device."